src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -4139,10 +4139,17 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     void handle_print(const T &x) {
         std::vector<llvm::Value *> args;
         std::vector<std::string> fmt;
-        for (size_t i=0; i<x.n_values; i++) {
+        // separator
+        this->visit_expr_wrapper(x.m_values[x.n_values - 1], true);
+        llvm::Value* separator = tmp;
+        for (size_t i=0; i<x.n_values - 1; i++) {
```

Comment:
This change effectively encodes the separator as the first argument to Print.

Let's add it as a new argument to `Print`, called `string separator`. That will make it clean. Also force the change in LFortran. This change here would break LFortran, since it is not using this new convention. As an argument, it would not break it.

We ultimately need a language independent representation of string formatting, but I think this little change (adding `separator` in ASR's Print) goes in the right direction and is clean enough.

Reply:
Sure!